File: quick_sort.cpp
#include <iostream>
#include <vector>

using std::vector;
using std::cout;
using std::endl;

void swapOwn(int* a, int* b){
   int t = *a;
   *a = *b;
   *b = t;
}

int pivot(int nums[], int low, int high){
   int pivot = nums[high];
   int i = (low - 1);

   for (int j = low; j <= high - 1; ++j){
      if (nums[j] < pivot){
         ++i;
         swapOwn(&nums[i], &nums[j]);
      }
   }
   swapOwn(&nums[i + 1], &nums[high]);
   return (i + 1);
}
void quickSort(int nums[], int left, int right){
   if (left < right){
      int pivotIndex = pivot(nums, left, right);
      quickSort(nums, left, pivotIndex - 1);
      quickSort(nums, pivotIndex + 1, right);
   }
}
int main(){
   int testNums[] = {12, 96, 3,  35, 2, 5, 1};
   quickSort(testNums, 0, 6);
   for (auto num : testNums)
      cout<<num<<" ";
   cout<<"\n";
}

File: quick_sort.js


function quickSort(nums) {
	if (nums.length <= 1) {
		return nums;
	}
	let pivot = nums[0];
	let left = [];
	let right = [];
	for (let i = 1; i < nums.length; i++) {
		nums[i] < pivot ? left.push(nums[i]) : right.push(nums[i]);
	}
	return quickSort(left).concat(pivot, quickSort(right));
}

let someArray = [93, 34, 35, 2, 214, 34, 5, 7]
for (let i = 0; i < someArray.length; i++) {
	console.log(Array[i]);
}

let sortedArray = quickSort(someArray);

for (let i = 0; i < sortedArray.length; i++) {
	console.log(sortedArray[i]);
}

